src/main/java/com/github/maleksandrowicz93/cqrsdemo/student/EditStudentDataCommandHandler.java

```diff
@@ -0,0 +1,30 @@
+package com.github.maleksandrowicz93.cqrsdemo.student;
+
+import com.github.maleksandrowicz93.cqrsdemo.student.dto.EditStudentDataCommand;
+import com.github.maleksandrowicz93.cqrsdemo.student.dto.StudentDto;
+import com.github.maleksandrowicz93.cqrsdemo.student.exception.InvalidCredentialsException;
+import com.github.maleksandrowicz93.cqrsdemo.student.exception.StudentNotFoundException;
+import lombok.RequiredArgsConstructor;
+
+@RequiredArgsConstructor
+class EditStudentDataCommandHandler {
+
+    private final StudentRepository studentRepository;
+
+    StudentDto handle(long studentId, EditStudentDataCommand command)
+            throws InvalidCredentialsException, StudentNotFoundException {
+        String email = command.getEmail();
+        if (email == null || email.isBlank()) {
+            throw new InvalidCredentialsException();
+        }
+        return studentRepository.findById(studentId)
+                .map(student -> {
```

Comment:
to mapowanie tutaj jest sztuczne, bo nie jest wykorzystywany `student` - zamiast tego wystarczyłoby bezpośrednio tutaj `orElseThrow`, żeby nie trzeba było doszukiwać się tej logiki - kod bardziej wprost pokazywałby, że nie korzysta z tego, co jest w bazie, a tylko sprawdza istnienie elementu